argo_workflow_tools/dsl/dag_compiler.py

```diff
@@ -76,18 +75,58 @@ def _create_task_script(
             "result", DefaultParameterBuilder(function_signature.return_annotation)
         )
         outputs = output_builder.variable_to_output("result", "result", func_obj)
-    call = f"result={func_obj.func.__name__}({str.join(',', inspect.signature(func_obj.func).parameters.keys())})"
+
+    func_code = _extract_source_without_decorators(func_obj.func)
+
+    pre_func_hook = func_obj.pre_func_hooks or _dummy_pre_func_hook
+    pre_func_hook_code = _extract_source_without_decorators(pre_func_hook)
+
+    post_func_hook = func_obj.post_func_hooks or _dummy_post_func_hook
+    post_func_hook_code = _extract_source_without_decorators(post_func_hook)
+
+    script = _build_full_source_script(func_obj, func_code, builder_imports, inputs, outputs, pre_func_hook,
+                                       pre_func_hook_code,  post_func_hook, post_func_hook_code)
+    return script
+
+
+def _build_full_source_script(func_obj, func_code, builder_imports, inputs, outputs,
+                              pre_func_hook,  pre_func_hook_code,  post_func_hook, post_func_hook_code):
+    indentation = ' ' * 4
+    func_call = f"result={func_obj.func.__name__}({str.join(',', inspect.signature(func_obj.func).parameters.keys())})"
+    func_and_hooks_call = (
+            f"{pre_func_hook.__name__}()\n"
+            + f"try:\n"
+            + f"{indentation}{func_call}\n"
```

Comment:
I'm not a huge fan of always rendering dummy funcs. the "space" of inline code is limited, and adding more code to the yaml might create problems when the pipelines or parameters grow. .

Reply:
the reason why I did it is that the rendering code is more complex than the rest, so I preferred not adding logic to it, you think that the space is worth more complicated code?